File: apps/divsoup/lib/divdump/analyzer/achievements/hyperlink_collector/platinum.ex
defmodule Divsoup.Achievement.HyperlinkCollectorPlatinum do
  alias Divsoup.Achievement
  alias Divsoup.Util.Hyperlink

  @behaviour Divsoup.AchievementRule
  
  # Minimum number of unique external domains required
  @min_domains 50

  @impl true
  def evaluate(html_tree, _) do
    # Count unique external domains
    domain_count = Hyperlink.count_unique_external_domains(html_tree)
    
    if domain_count >= @min_domains do
      []
    else
      ["The page only links to #{domain_count} different external domains, needs at least #{@min_domains}"]
    end
  end

  @impl true
  def achievement() do
    %Achievement{
      hierarchy: :platinum,
      title: "Hyperlink Connoisseur",
      group: "hyperlink_collector",
      description: "The page links to at least #{@min_domains} different external domains"
    }
  end
end

File: src/achievements/hyperlink_collector/platinum.ts
import type { AchievementRule } from "../types";
import { countUniqueExternalDomains } from "../utils";

const minimumExternalDomains = 50;

export const rule: AchievementRule = {
  id: "hyperlink_collector.platinum",
  title: "Hyperlink Connoisseur",
  group: "hyperlink_collector",
  description: `Page contains links to at least <strong>${minimumExternalDomains}</strong> different external domains`,
  hierarchy: "platinum",
  evaluate: ({ doc }) => countUniqueExternalDomains(doc) >= minimumExternalDomains,
};
